core/src/main/java/de/bwaldvogel/mongo/backend/Utils.java

```diff
@@ -260,6 +260,24 @@ static boolean hasSubdocumentValue(Object document, String key) {
         }
     }
 
+    static boolean canFullyTraverseSubkeyForRename(Object document, String key) {
```

Comment:
I don't entirely love needing this, but it seems renames are a little special. If you have alternative ideas, I'm all ears.

Reply:
I don’t have a good idea either at the moment. Lets keep it like that. We can refactor it at any time and this looks like a clear improvement!